src/javaharness/java/arcs/webimpl/PortableJsonJsImpl.java

```diff
@@ -10,10 +10,56 @@
 import java.util.function.Consumer;
 import java.util.stream.Collectors;
 import jsinterop.base.Any;
+import jsinterop.base.Js;
+import jsinterop.annotations.JsMethod;
+import jsinterop.annotations.JsFunction;
 
 public class PortableJsonJsImpl implements PortableJson {
     private Any jsonObj;
 
+    @JsMethod(namespace="<window>", name="eval")
+    private native static Object eval(String js);
+
+    @JsFunction
+    interface Getter {
+      Object get(Object obj, String key);
+    }
+
+    @JsFunction
+    interface HasChecker {
+        boolean has(Object obj, String key);
+    }
+
+    @JsFunction
+    interface Setter {
+      Object set(Object obj, String key, Object value);
+    }
+
+    @JsFunction
+    interface Keys {
+        String[] get(Object obj);
+    }
+
+    private <T> T getValue(String key) {
+      Getter getter = (Getter) eval("(obj, key) => { return obj[key]; }");
+      return Js.uncheckedCast(getter.get(jsonObj, key));
+    }
+
+    private boolean hasStringKey(String key) {
+      HasChecker hasChecker = (HasChecker) eval("(obj, key) => { return key in obj; }");
+      return hasChecker.has(jsonObj, key);
+    }
+
+    private <T> T setValue(String key, T value) {
+        Setter setter = (Setter) eval("(obj, key, value) => { obj[key] = value; };");
+        return Js.uncheckedCast(setter.set(jsonObj, key, value));
+    }
+
+    private List<String> getAllKeys() {
+        Keys keys = (Keys) eval("let keys = []; (obj) => { for (let key in obj) { keys.push(key); } return keys; }");
+        return Arrays.asList(keys.get(jsonObj));
```

Comment:
This is a dangerous method, it might work, but isn't recommended because a primitive array of JS strings isn't the same as an array of String[] (that is 'x instanceof String[]' will fail). As long as there's no cast check inserted by the compiler, it's work. 

A better technique would be to return JsArrayLike (https://github.com/google/jsinterop-base/blob/master/java/jsinterop/base/JsArrayLike.java), but this might fail for the same reasons JsPropertyMap is failing. 

I think this is ok for now (similar technique is used in JsArrayLike), but we'll need to fix all of this soon since it might fail for random reasons in the future.

Reply:
i do hope the upstream gets fixed soon and we can go back to doing everything the normal way (did you email that team? could you maybe, cc me on that thread? thanks)